tools/EventsReader/SqlConnectionBuilder.cs

```diff
@@ -29,5 +29,13 @@ public async Task<SqlConnection> GetSqlConnectionAsync(string initialCatalog = n
             await Task.CompletedTask;
             return new SqlConnection(_connectionString);
         }
+
+        public async Task<SqlConnection> GetReadOnlySqlConnectionAsync(string initialCatalog = null, int? maxPoolSize = null, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            var builder = new SqlConnectionStringBuilder(_connectionString);
+            builder.ApplicationIntent = ApplicationIntent.ReadOnly;
+            return new SqlConnection(builder.ToString());
```

Comment:
You can use a Task.FromResult instead of having the await Task.CompletedTask.

Reply:
It should be functionally identical. I tested this code, and suggest to leave it as is.